### PR TITLE
Simple Get (v2)

### PR DESCRIPTION
**Pull request by xray**
A far more test driven approach to Simple Get.

### CODE REVIEW

File: src/ProtocolHandler/DataParser/Request.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            Protocol = protocol;
+            Headers = headers;
+        }
+
+        public bool Equals(Request other)
+        {
+             var requestLineValuesEqual = Method == other.Method &&
+                   Path == other.Path &&
+                   Protocol == other.Protocol;
+             if (!requestLineValuesEqual) return false;
```
Comment (ssunday) [inline src/ProtocolHandler/DataParser/Request.cs]: Could this be extracted out? (I mean the var)

File: src/ProtocolHandler/ResponseRetriever/ResRetriever.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        public Response Retrieve(Request req)
+        {
+            if (req.Path == "/simple_get")
+            {
+                return new Response("HTTP/1.1", 200, "OK")
+                    .AddHeader("Date", $"{DateTimeProvider.Now():R}");
+            }
+
+            return new Response("HTTP/1.1", 404, "Not Found")
+                .AddHeader("Date", $"{DateTimeProvider.Now():R}");
```
Comment (ssunday) [inline src/ProtocolHandler/ResponseRetriever/ResRetriever.cs]: The date parts are identical. Anyway to de-dupe that?

File: src/ProtocolHandler/DataParser/Request.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
+using System.Text;
+using Chorizo.ProtocolHandler.HttpHeaders;
+
+namespace Chorizo.ProtocolHandler.DataParser
```
Comment (kevbuchanan) [inline src/ProtocolHandler/DataParser/Request.cs]: Request probably doesn't belong in this namespace.

File: src/ProtocolHandler/DataParser/Request.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        {
+            _method = method;
+            _path = path;
+            _protocol = protocol;
+            _headers = headers;
+        }
+
+        public string Method()
+        {
+            return _method;
```
Comment (kevbuchanan) [inline src/ProtocolHandler/DataParser/Request.cs]: Can this just be a property named `Method`?

File: src/ProtocolHandler/DataParser/RequestParser.cs [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            var requestLine = requestString.Split("\r\n", StringSplitOptions.RemoveEmptyEntries)[0];
+            return requestLine.Split(" ");
+        }
+
+        private static Headers GetRequestHeaders(string requestString)
+        {
+            var headers = new Headers();
+            var requestLines = requestString.Split("\r\n", StringSplitOptions.RemoveEmptyEntries);
+            var headerLines = requestLines.Skip(1);
+            Console.WriteLine($"this: {requestString}");
```
Comment (kevbuchanan) [inline src/ProtocolHandler/DataParser/RequestParser.cs]: Remove this.

File: src/ProtocolHandler/HttpHeaders/Header.cs [hunk 6]
```diff
@@ -0,0 +1,37 @@
+using System;
+
+namespace Chorizo.ProtocolHandler.HttpHeaders
```
Comment (kevbuchanan) [inline src/ProtocolHandler/HttpHeaders/Header.cs]: Probably doesn't need to be in the ProtocolHandler namespace.

File: src/ProtocolHandler/ResponseRetriever/ResRetriever.cs [hunk 7]
```diff
@@ -0,0 +1,27 @@
+using Chorizo.Logger;
+using Chorizo.ProtocolHandler.DataParser;
+
+namespace Chorizo.ProtocolHandler.ResponseRetriever
+{
+    public class ResRetriever:IResponseRetriever
```
Comment (kevbuchanan) [inline src/ProtocolHandler/ResponseRetriever/ResRetriever.cs]: Naming could be a lot better.

File: src/ProtocolHandler/ResponseRetriever/ResRetriever.cs [hunk 8]
```diff
@@ -0,0 +1,27 @@
+using Chorizo.Logger;
+using Chorizo.ProtocolHandler.DataParser;
+
+namespace Chorizo.ProtocolHandler.ResponseRetriever
```
Comment (kevbuchanan) [inline src/ProtocolHandler/ResponseRetriever/ResRetriever.cs]: I don't think this needs to be in the ProtocolHandler namespace.

File: src/ProtocolHandler/HttpHeaders/Header.cs (hunk 6 above)
Comment (xray) [reply src/ProtocolHandler/HttpHeaders/Header.cs]: Added to Exchange namespace based on this [StackOverflow](https://stackoverflow.com/a/325358) post.

File: src/ProtocolHandler/DataParser/Request.cs (hunk 3 above)
Comment (xray) [reply src/ProtocolHandler/DataParser/Request.cs]: Added to Exchange namespace based on this [StackOverflow](https://stackoverflow.com/a/325358) post.

**Pull request merged**